Makefile

```diff
@@ -14,7 +14,9 @@ run-postgres:
 	$(CONTAINER_RUNTIME) run -it $(POSTGRES_RUN_ARGS) -v data:/var/lib/postgresql/data -e POSTGRES_USER=kai -e POSTGRES_PASSWORD=dog8code -e POSTGRES_DB=kai -p 5432:5432 docker.io/library/postgres:16.3
 
 run-server:
-	PYTHONPATH=$(KAI_PYTHON_PATH) python kai/server.py
+	bash -c 'set -m; _trap () { kill -15 $$PID; } ; trap _trap SIGINT ;\
+	PYTHONPATH=$(KAI_PYTHON_PATH) python kai/server.py & export PID=$$! ;\
+	while kill -0 $$PID > /dev/null 2>&1; do wait $$PID; done'
```

Comment:
"kill -0" sends no signal; it just checks for the pids existence. This loop just prevents the shell from exiting before the graceful shutdown is complete.